File: api/py/day1.py
from http.server import BaseHTTPRequestHandler
from datetime import datetime
import json

class handler(BaseHTTPRequestHandler):

  def do_POST(self):
    content_len = int(self.headers.get('Content-Length'))
    post_body = self.rfile.read(content_len)
    body = json.loads(post_body)
    data = body['input']
    if data:
        numbers = list(map(int, data.split(",")))
        silver = 0
        gold = 0
        for i in numbers:
            for j in numbers:
                if i + j == 2020:
                    silver = i * j
                for k in numbers:
                    if i + j + k == 2020:
                        gold = i * j * k


        result = { "silver": str(silver), "gold": str(gold) }
    else:
        result = { "silver": "No input", "gold": "No input" }
    
    self.send_response(200)
    self.send_header('Content-Type', 'application/json')
    self.end_headers()
    self.wfile.write(json.dumps(result).encode('utf_8'))
    return

File: api/ts/day1.ts
import { NowRequest, NowResponse } from "@vercel/node";
import { StarResult } from "./shared/data";

export default (request: NowRequest, response: NowResponse) => {
  let input = request.body && request.body.input;
  let responseMessage: StarResult;

  const solve = (): StarResult => {
    if (input) {
      const numbers = input.split(";").map((str: string) => parseInt(str));
      let silver = 0;
      let gold = 0;

      for (let i = 0; i < numbers.length; i++) {
        for (let j = 0; j < numbers.length; j++) {
          if (numbers[i] + numbers[j] == 2020) {
            silver = numbers[i] * numbers[j];
          }
          for (let k = 0; k < numbers.length; k++) {
            if (numbers[i] + numbers[j] + numbers[k] == 2020) {
              gold = numbers[i] * numbers[j] * numbers[k];
            }
          }
        }
      }
      return { silver: silver.toString(), gold: gold.toString() };
    } else return { silver: "No input", gold: "No input" };
  };

  responseMessage = solve();
  response.status(200).json(responseMessage);
};
